app/assets/javascripts/pretty-text/engines/discourse-markdown/watched-words-replace.js

```diff
@@ -10,9 +10,17 @@ function findAllMatches(text, matchers, useRegExp) {
   const matches = [];
 
   if (useRegExp) {
+    const maxMatches = 100;
+    let index = 0;
+
     matchers.forEach((matcher) => {
       let match;
-      while ((match = matcher.pattern.exec(text)) !== null) {
+      while (
+        (match = matcher.pattern.exec(text)) !== null &&
+        index < maxMatches
```

Comment:
Tricky but you can do this as well

```suggestion
        ++index < maxMatches
```